woocommerce/woocommerce-ios
# Analytics for local notifications

Part of #9665 


## Description

This PR adds analytics for scheduling/canceling local notifications as well as interactions with them:



Event | Trigger | Properties
-- | -- | --
*_local_notification_tapped | When a notification is tapped. | type:  store_creation_complete \|  one_day_after_store_creation_name_without_free_trial \|  one_day_before_free_trial_expires \| one_day_after_free_trial_expires
*_local_notification_dismissed | When a notification is dismissed. | type:  store_creation_complete \|  one_day_after_store_creation_name_without_free_trial \|  one_day_before_free_trial_expires \| one_day_after_free_trial_expires
*_local_notification_scheduled | When a notification is scheduled. | type:  store_creation_complete \|  one_day_after_store_creation_name_without_free_trial \|  one_day_before_free_trial_expires \| one_day_after_free_trial_expires
*_local_notification_canceled | When a notification is manually canceled (e.g when a store is upgraded or upon logout) | type:  store_creation_complete \|  one_day_after_store_creation_name_without_free_trial \|  one_day_before_free_trial_expires \| one_day_after_free_trial_expires



## Testing instructions


We can test any of the scenarios of WooExpress notifications. Below are instructions for scenario 3:

Prerequisites: since the remote feature flags are still not deployed and remain false, some code changes are required to test this PR:

 - In LocalNotificationScheduler.schedule, remove the whole condition on isRemoteFeatureFlagEnabled(remoteFeatureFlag) so that the remote feature flag isn't used
- Please also ensure you have access to a site running a trial plan that is not expired yet for at least 2 days.

---

- Launch the app, log in, and switch to your test site if needed. For easier testing, please run the app on a physical device.
- In Xcode console, notice `🔵 Tracked local_notification_scheduled, properties: [AnyHashable("blog_id"): <blog_id>, AnyHashable("type"): "one_day_before_free_trial_expires", AnyHashable("is_wpcom_store"): true]`
- Send the app to background, open Settings, and change the time of the device to 1 day and 1 minute prior to the expiration date. The expiry date is at 0:00, so my expiry date is on June 1, then I should change the date to May 30, 23:59.
- Wait for 1 minute, then you should see a notification about the expiring trial plan. Tap on the notification, notice in Xcode console: `🔵 Tracked local_notification_tapped, properties: [AnyHashable("blog_id"): <blog_id>, AnyHashable("is_wpcom_store"): true, AnyHashable("type"): "one_day_before_free_trial_expires"]`
- Reset the device time, open the app, and log out of your account. Notice in Xcode console: `🔵 Tracked local_notification_canceled, properties: [AnyHashable("blog_id"): <blog_id>, AnyHashable("type"): "one_day_after_free_trial_expires", AnyHashable("is_wpcom_store"): true]`

## Screenshots

N/A

---
